egs/wsj/s5/utils/parallel/slurm.pl

```diff
@@ -1,628 +1,391 @@
+
+PARSE_ARGS:
+while (@ARGV >= 2) {
+  my $arg = $ARGV[0];
+
+  # Switches.
+  if ($arg =~ m/^-/) {
     my $switch = shift @ARGV;
+    fatal "invalid switch '$arg'; single-dash switches are unsupported!"
+        unless $arg =~ m/^--/;
 
-    if ($switch eq "-V") {
-      $qsub_opts .= "-V ";
-    } else {
-      my $argument = shift @ARGV;
-      if ($argument =~ m/^--/) {
-        print STDERR "WARNING: suspicious argument '$argument' to $switch; starts with '-'\n";
-      }
-      if ($switch eq "-sync" && $argument =~ m/^[yY]/) {
-        $sync = 1;
-        $qsub_opts .= "$switch $argument ";
-      } elsif ($switch eq "-pe") { # e.g. -pe smp 5
-        my $argument2 = shift @ARGV;
-        $qsub_opts .= "$switch $argument $argument2 ";
-        $num_threads = $argument2;
-      } elsif ($switch =~ m/^--/) { # Config options
-        # Convert CLI option to variable name
-        # by removing '--' from the switch and replacing any
-        # '-' with a '_'
-        $switch =~ s/^--//;
-        $switch =~ s/-/_/g;
-        $cli_options{$switch} = $argument;
-      } else {  # Other qsub options - passed as is
-        $qsub_opts .= "$switch $argument ";
-      }
-    }
+    # '--' ends this script switch parse.
+    last PARSE_ARGS if $arg eq '--';
+
+    # Convert CLI option to variable name by removing '--' from the switch and
+    # replacing any '-' with a '_' ('--max-jobs-run' => 'max_jobs_run').
+    $switch =~ s/^--//;
+    $switch =~ s/-/_/g;
+
+    my $swarg = shift @ARGV;
+    $cli_options{$switch} = $swarg;
+
+    $debug = $swarg if ($switch eq 'debug');
+
+    next PARSE_ARGS;
   }
-  if ($ARGV[0] =~ m/^([\w_][\w\d_]*)+=(\d+):(\d+)$/) { # e.g. JOB=1:20
-    $array_job = 1;
-    $jobname = $1;
-    $jobstart = $2;
-    $jobend = $3;
-    shift;
-    if ($jobstart > $jobend) {
-      die "$0: invalid job range $ARGV[0]";
-    }
-    if ($jobstart <= 0) {
-      die "$0: invalid job range $ARGV[0], start must be strictly positive (this is a GridEngine limitation).";
-    }
-  } elsif ($ARGV[0] =~ m/^([\w_][\w\d_]*)+=(\d+)$/) { # e.g. JOB=1.
-    $array_job = 1;
-    $jobname = $1;
+
+  # Job spec, JOB=1:42 or JOB=1
+  if ($arg =~ m/^([\pL_][\pL\d_]*) = (\d+)(?: :(\d+) )?$/ax) {
+    fatal "only one job spec is allowed: '$arg'" if $jobvar;
+    $jobvar = $1;
     $jobstart = $2;
-    $jobend = $2;
+    $jobend = $3 // $2;  # Treat 'X=42' as if it were 'X=42:42'.
+    fatal "invalid job range '$arg': start > end" if $jobstart > $jobend;
+    fatal "invalid job range '$arg': must be positive" if $jobstart == 0;
     shift;
-  } elsif ($ARGV[0] =~ m/.+\=.*\:.*$/) {
-    print STDERR "Warning: suspicious first argument to $0: $ARGV[0]\n";
+    next PARSE_ARGS;
   }
-}
 
-if (@ARGV < 2) {
-  print_usage();
+  $arg =~ m/.=.*:/ and
+      warning "suspicious argument '$arg', malformed job spec?";
+
+  last PARSE_ARGS;
 }
 
-if (exists $cli_options{"config"}) {
-  $config = $cli_options{"config"};
```

Comment:
in the original slurm, the $config is pre-set to "conf/slurm.conf", which results in automatically reading the config _if_ it exists. It is also the default behavior of queue.pl, so it might be preferred to stick with that behavior also...